src/mca/odls/base/odls_base_default_fns.c

```diff
@@ -1584,6 +1570,25 @@ void prrte_odls_base_default_wait_local_proc(int fd, short sd, void *cbdata)
         goto MOVEON;
     }
 
+    prrte_node_t *nptr = NULL; bool removed_proc = false;
+    for(int i = 0; i < jobdat -> map -> num_nodes; i++) {
+        if (NULL == (nptr = (prrte_node_t*)prrte_pointer_array_get_item(jobdat->map->nodes, i))) {
+            continue;
+        }
+        for(int j = 0; j < (int) nptr -> procs -> size; j++) {
```

Comment:
Why are we doing this here? The state component is supposed to recover resources, not the odls. Is it not doing it correctly?

Reply:
I'll check - maybe this isn't needed or not in the right spot.

Reply:
I removed it, node -> next_node_rank just needed to be decremented.